modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -925,9 +993,18 @@ private CompletableFuture<Object> processSingleEntryAction(ReadWriteSingleRowRep
                 CompletableFuture<RowId> lockFut = takeLocksForUpsert(searchKey, indexId, txId);
 
                 return lockFut.thenCompose(lockedRowId -> {
-                    CompletableFuture raftFut =
-                            lockedRowId != null ? applyCmdWithExceptionHandling(new UpdateCommand(lockedRowId, searchRow, txId)) :
-                                    applyCmdWithExceptionHandling(new UpdateCommand(new RowId(partId), searchRow, txId));
+                    RowIdMessage rowIdMsg = lockedRowId != null ? RowIdMessage.fromRowId(msgFactory, lockedRowId) :
+                            msgFactory.rowIdMessage()
+                                    .partitionId((short) partId)
+                                    .uuid(Timestamp.nextVersion().toUuid())
+                                    .build();
+
+                    CompletableFuture raftFut = applyCmdWithExceptionHandling(
```

Comment:
Please add type arguments to raw types when you modify the code. It'll make it better

Reply:
I had to add very generic CompletableFuture<Object> - to me it is not very useful but at least it eliminated IDE warnings.